spec/controllers/admin/sponsorships_controller_spec.rb

```diff
@@ -3,30 +3,74 @@
 
 describe Admin::SponsorshipsController, type: :controller do
 
-  let(:orphan) { instance_double Orphan }
   let(:sponsor) { instance_double Sponsor }
-  let(:sponsorship) { instance_double Sponsorship, sponsor: sponsor, orphan: orphan }
+  let(:sponsorship) { instance_double Sponsorship }
 
   before(:each) do
     sign_in instance_double(AdminUser)
-    allow(sponsor).to receive(:id).and_return(1)
-    allow(Sponsor).to receive(:find).with('1').and_return(sponsor)
-    allow(Sponsorship).to receive(:find).with('1').and_return(sponsorship)
+    allow(Sponsor).to receive(:find).with('1').and_return sponsor
   end
 
   describe 'inactivate' do
-    describe 'is successful' do
+
+    before(:each) do
+      expect(sponsor).to receive_message_chain(:sponsorships, :find)
+    end
+
+    context 'when successful' do
+
+      before(:each) do
+        expect(InactivateSponsorship).to receive_message_chain(:new, :call).
+          and_return true
+        put :inactivate, { id: 1, sponsor_id: 1, end_date: '1-1-2017' }
+      end
+
+      it 'sets flash[:success] message' do
+        expect(flash[:success]).not_to be_nil
+        expect(flash[:warning]).to be_nil
+      end
+
+      it 'redirects to sponsor show view' do
+        expect(response).to redirect_to admin_sponsor_path(sponsor)
+      end
+    end
+
+    context 'when unsuccessful' do
+
       before(:each) do
-        allow(sponsorship).to receive(:inactivate).and_return(sponsorship)
-        put :inactivate, { id: 1, sponsor_id: 1, end_date: "1-1-2017" }
-      end  
+        expect(InactivateSponsorship).to receive_message_chain(:new, :call).
+          and_raise('BOOM')
+        put :inactivate, { id: 1, sponsor_id: 1, end_date: '1-1-2017' }
+      end
+
+      it 'sets flash[:warning] message' do
+        expect(flash[:success]).to be_nil
+        expect(flash[:warning]).to eq 'BOOM'
+      end
+
+      it 'redirects to sponsor show view' do
+        expect(response).to redirect_to admin_sponsor_path(sponsor)
+      end
+    end
+  end
+
+  describe 'destroy' do
+
+    before(:each) do
+      allow(sponsor).to receive_message_chain(:sponsorships, :find)
+    end
+
+    context 'when successful' do
 
-      it 'calls #inactivate on sponsorship' do
-        expect(sponsorship).to have_received :inactivate
+      before(:each) do
+        expect(DestroySponsorship).to receive_message_chain(:new, :call).
+          and_return true
+        delete :destroy, { id: 1, sponsor_id: 1 }
```

Comment:
The only param expected by the `DestroySponsorship` service is sponsorship. We can leave the route as is in AA and implement it more efficiently in Rails without having to refactor the service.